src/execution.ts

```diff
@@ -1637,3 +1668,214 @@ function getParentArgIndexes(context: CompilationContext) {
+    // the GraphQL specification. The `execute` function provides the
+    // "ExecuteSubscriptionEvent" algorithm, as it is nearly identical to the
+    // "ExecuteQuery" algorithm, for which `execute` is also used.
+    // We use our `query` function in place of `execute`
+    const mapSourceToResponse = (payload: any) =>
+      queryFn(payload, executionContext.context, executionContext.variables);
+
+    return mapAsyncIterator(resultOrStream, mapSourceToResponse);
+  };
+}
+
+function createBoundSubscribe(
+  compilationContext: CompilationContext,
+  document: DocumentNode,
+  func: (
+    context: ExecutionContext
+  ) => Promise<AsyncIterableIterator<ExecutionResult> | ExecutionResult>,
+  getVariableValues: (inputs: { [key: string]: any }) => CoercedVariableValues,
+  operationName: string | undefined
+): CompiledQuery["subscribe"] {
+  const {
+    resolvers,
+    typeResolvers,
+    isTypeOfs,
+    serializers,
+    resolveInfos
+  } = compilationContext;
+  const trimmer = createNullTrimmer(compilationContext);
+  const fnName = operationName ? operationName : "subscribe";
+
+  const ret = {
+    async [fnName](
+      rootValue: any,
+      context: any,
+      variables: Maybe<{ [key: string]: any }>
+    ): Promise<AsyncIterableIterator<ExecutionResult> | ExecutionResult> {
+      // this can be shared across in a batch request
+      const parsedVariables = getVariableValues(variables || {});
+
+      // Return early errors if variable coercing failed.
+      if (failToParseVariables(parsedVariables)) {
+        return { errors: parsedVariables.errors };
+      }
+
+      const executionContext: ExecutionContext = {
+        rootValue,
+        context,
+        variables: parsedVariables.coerced,
+        safeMap,
+        inspect,
+        GraphQLError: GraphqlJitError,
+        resolvers,
+        typeResolvers,
+        isTypeOfs,
+        serializers,
+        resolveInfos,
+        trimmer,
+        promiseCounter: 0,
+        nullErrors: [],
+        errors: [],
+        data: {}
+      };
+
+      return func.call(null, executionContext);
+    }
+  };
+
+  return ret[fnName];
+}
+
+/**
+ * Given an AsyncIterable and a callback function, return an AsyncIterator
+ * which produces values mapped via calling the callback function.
+ */
+function mapAsyncIterator<T, U, R = undefined>(
+  iterable: AsyncGenerator<T, R, undefined> | AsyncIterable<T>,
+  callback: (value: T) => U | Promise<U>
+): AsyncGenerator<U, R, undefined> {
+  const iterator = iterable[Symbol.asyncIterator]();
+
+  async function mapResult(
+    result: IteratorResult<T, R>
+  ): Promise<IteratorResult<U, R>> {
+    if (result.done) {
+      return result;
+    }
+
+    try {
+      return { value: await callback(result.value), done: false };
+    } catch (error) {
+      if (typeof iterator.return === "function") {
+        try {
+          await iterator.return();
+        } catch (e) {
+          /* ignore error */
```

Comment:
Would it be possible to extend the comment here to explicitly state why we ignore the error?